### ISSUE TITLE
Helpers for producing JDBC timestamps

### ISSUE THREAD

**Issue by michaelklishin** [clj-time/clj-time#149]
At the moment, to produce a timestamp that can be inserted into a relational database using JDBC (clojure.java.jdbc or any alternative), it takes to

 * Use `clj-time.core/date-time` or similar to create a Joda Time `DateTime`.
 * Convert it to `java.util.Date`.
 * Convert it to `java.sql.Timestamp`

Because there is no easy way extend JDBC (or its Clojure clients) to support arbitrary types, I think
we should provide a few common helpers that produce `Timestamp` instances. For example,
`clj-time.sql/date-time` could work like `clj-time.core/date-time` does but do the extra conversion.

@seancorfield what do you think?

**Comment by seancorfield** [comment]
Any thoughts on how to provide safe, optional protocol extensions in clj-time that allow java.jdbc's protocols to support this?

**Comment by gws** [comment]
I'm not sure if this is necessarily the most robust approach but it works well for me against PostgreSQL, I haven't tested it against other databases though, though there's nothing PostgreSQL-specific in here.

It's basically what @michaelklishin already mentioned - putting this in a ns and requiring it somewhere:

```clojure
(ns my.jdbc.ext
  (:require [clj-time.coerce :as time-coerce]
            [clojure.java.jdbc :as jdbc]))

; http://clojure.github.io/java.jdbc/#clojure.java.jdbc/IResultSetReadColumn
(extend-protocol jdbc/IResultSetReadColumn
  java.sql.Timestamp
  (result-set-read-column [col _ _]
    (time-coerce/from-sql-time col)))

; http://clojure.github.io/java.jdbc/#clojure.java.jdbc/ISQLValue
(extend-protocol jdbc/ISQLValue
  org.joda.time.DateTime
  (sql-value [v]
    (time-coerce/to-sql-time v)))
```

**Comment by michaelklishin** [comment]
@gws feel free to submit a PR. I'd change `time-coerce` to `tc` but otherwise it looks reasonable.

**Comment by michaelklishin** [comment]
@gws a namespace I'd pick is `clj-time.jdbc`.

**Comment by seancorfield** [comment]
Fixed by #176
